### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by szepeviktor**
- :zero: Byte-level
  - Look for non-ASCII characters
    (non-English alphabets, whitespace characters, control characters)
- :one: Syntax errors
  - Check source code for syntax errors
- :two: Run unit and functional tests
- :three: Static Analysis
  - Run static analysis: **PHPStan**
- :four: Coding Standards
  - Check coding style
  - Adhere to EditorConfig

### CODE REVIEW

**Comment by szepeviktor** [comment]
I'm in love with this Beauty.

kép


@DarkGhostHunter Should I fix CS problems?

**Comment by szepeviktor** [comment]
> Should I fix CS problems?

Done 🍏

**Comment by szepeviktor** [comment]
@DarkGhostHunter Please look at `TValue` in src/Refresh.php once more.

**Comment by DarkGhostHunter** [comment]
I see some commented code on the test runs. Care to say what are their purpose?

File: .github/workflows/integrate.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php-version }}"
+
+      - name: "Checkout code"
+        uses: "actions/checkout@v3"
+
+#      - name: "Set Laravel version"
+#        run: |
+#          composer require --no-progress --no-update "laravel/framework:${{ matrix.laravel }}"
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml]: Currently we have 1 supported Laravel framework version.
So it needs no extra action.

File: .github/workflows/integrate.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        with:
+          dependency-versions: "${{ matrix.dependencies }}"
+
+      - name: "Execute unit tests"
+        run: "composer run-script test"
+        # @TODO Functional tests
+
+#      - name: "Tests have failed: display logs"
+#        if: "${{ failure() }}"
+#        run: "cat storage/logs/*.log"
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml]: It there is a log file with stacktrace  etc CI can display that.

**Comment by szepeviktor** [comment]
> Care to say what are their purpose?

Comments added.

**Comment by DarkGhostHunter** [comment]
> @DarkGhostHunter Please look at `TValue` in src/Refresh.php once more.

The callback will receive `TValue` (the value exists) or `null` (cache miss), so the PHPDoc should reflect that always.

**Comment by szepeviktor** [comment]
As `CacheRefresh` is a library there could be a CI job that installs it into Laravel sample application and does some real-world testing.

**Comment by szepeviktor** [comment]
> The callback will receive `TValue` (the value exists) or `null` (cache miss), so the PHPDoc should reflect that always.

Done.

What is the second parameter's (`Laragear\CacheRefresh\Expire`) default value?

**Comment by DarkGhostHunter** [comment]
> What is the second parameter's (`Laragear\CacheRefresh\Expire`) default value?

The object uses the TTL passed down, which is `null` by default.

```php
use Illuminate\Support\Facades\Cache;
use Illuminate\Support\Collection;
use Laragear\CacheRefresh\Expire;

Cache::refresh('rofl', function (?Collection $collection, Expire $expire): Collection {
   // ..
}
```

File: .github/workflows/integrate.yml (hunk 2 above)
Comment (DarkGhostHunter) [reply .github/workflows/integrate.yml]: Nice! It should be enabled, but I figured out that enabling `APP_DEBUG` in tests can also show stack traces when exceptions are thrown.

**Comment by szepeviktor** [comment]
@DarkGhostHunter Finally done 🍏

**Comment by DarkGhostHunter** [comment]
Thanks. I'm testing something on Rut 2.0, I'll get back to this tomorrow.

**Comment by DarkGhostHunter** [comment]
Made some changes: Changed to ubuntu-latest, removes redundant Code Standard.

**Comment by szepeviktor** [comment]
> removes redundant Code Standard

@DarkGhostHunter Does PHP CS Fixer run elsewhere?

The removed job contained a step to check EditorConfig adherence.

**Comment by DarkGhostHunter** [comment]
Style fixes are done already by StyleCI. 

I mistakenly deleted the PHP CS Fixer step, I’ll re add it later. 

Italo Baeza C.

**Comment by szepeviktor** [comment]
StyleCI is an external service. Log in and enable it for this repo: https://styleci.io/
https://github.com/Laragear/CacheRefresh/blob/1.x/.styleci.yml

**Comment by DarkGhostHunter** [comment]
Done. The PHP CS Fixer is not needed anyway. Any hiccup will be catch by PHP Stan.

**Pull request merged**